File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if nums[i]-1 == i:
                continue
            elif nums[i] == nums[nums[i]-1]:
                return nums[i]
            else:
                tmp = nums[nums[i]-1]
                nums[nums[i]-1] = nums[i]
                nums[i] = tmp
        
        return -1

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int zeroCounter=0, prod=1;

        for(int i : nums){
            if(i == 0){
                zeroCounter++;
            }else{
                prod *= i;
            }
        }
        if(zeroCounter > 1){
            return new int[nums.length];
        }
        for(int i=0; i<nums.length; i++){
            if(zeroCounter == 1){
                nums[i] = nums[i] == 0 ? prod : 0;
            }else{
                nums[i] = prod/nums[i];
            }
        }
        return nums;
    }
}  
